src/reducers/dependencies.reducer.js

```diff
@@ -120,6 +126,91 @@ export default (state: State = initialState, action: Action) => {
       });
     }
 
+    case ADD_DEPENDENCIES_START: {
+      const { projectId, dependencies } = action;
+
+      return produce(state, draftState => {
+        // this action name is a bit of a misnomer, since this block
+        // is used for both adding AND updating dependencies (otherwise,
+        // we wouldn't be able to batch installs/updates with a single
+        // `npm install` command
```

Comment:
Hm, it's not clear to me that we need to make a distinction between installing and updating... what's stopping us from treating updates as new installs? We should have all the dependency info from the action, and we can just overwrite existing entries with the new one, right?

Reply:
Updates are functionally treated like new installs, hence being included in the same block, but I think a distinction needs to be made to determine whether or not to display the DependencyInstalling component while this action flow is taking place.